src/layout/layout.js

```diff
@@ -170,23 +175,98 @@
+      evt.preventDefault();
+      // Send mousedown and mouseup to trigger ripple.
+      var e = new MouseEvent('mousedown');
+      evt.target.dispatchEvent(e);
+      e = new MouseEvent('mouseup');
+      evt.target.dispatchEvent(e);
+      // Send click.
+      evt.target.click();
+    } else if (evt.keyCode === this.Keycodes_.HOME && this.drawer_.classList.contains(this.CssClasses_.IS_DRAWER_OPEN)) {
+      this.drawerMenuItems_[0].focus();
+    } else if (evt.keyCode === this.Keycodes_.END && this.drawer_.classList.contains(this.CssClasses_.IS_DRAWER_OPEN)) {
+      this.drawerMenuItems_[this.drawerMenuItems_.length - 1].focus();
     }
   };
 
   /**
+   * Handles a keyboard navigation events on the drawer.
+   *
+   * @param {Event} evt The event that fired.
+   * @private
+   */
+  MaterialLayout.prototype.handleMenuNavKeyboardEvent_ = function(evt) {
+      // Only react when a draw exists, is open and tab is pressed.
+      if ((evt.keyCode === this.Keycodes_.TAB || evt.keyCode === this.Keycodes_.DOWN_ARROW ||
+            evt.keyCode === this.Keycodes_.UP_ARROW) && this.drawer_ &&
+            this.drawer_.classList.contains(this.CssClasses_.IS_DRAWER_OPEN)) {
+        var currentIndex = Array.prototype.slice.call(this.drawerMenuItems_).indexOf(evt.target);
+        // Tab forward and backward through navigation links. Collapse drawer
+        // when user reaches the end of the links.
+        if ((evt.shiftKey && evt.keyCode === this.Keycodes_.TAB) || evt.keyCode === this.Keycodes_.UP_ARROW) {
+          evt.preventDefault();
+          if (0 < currentIndex) {
+            this.drawerMenuItems_[currentIndex - 1].focus();
+          } else {
+            // This would loop to the top of the menu but blocks the user
+            // from accessing the browser itself. Instead I've opted to
+            // close the drawer and continue in the tab order.
+            // this.drawerMenuItems_[this.drawerMenuItems_.length - 1].focus();
+            this.toggleDrawer();
+          }
+        } else {
+          evt.preventDefault();
+          if (this.drawerMenuItems_.length > currentIndex + 1) {
+            this.drawerMenuItems_[currentIndex + 1].focus();
+          } else {
+            // This would loop to the top of the menu but blocks the user
+            // from accessing the browser itself. Instead I've opted to
+            // close the drawer and continue in the tab order.
+            // this.drawerMenuItems_[0].focus();
+            this.toggleDrawer();
+          }
+        }
+      }
+    };
+
+  /**
    * Handles changes in screen size.
    *
    * @private
    */
   MaterialLayout.prototype.screenSizeHandler_ = function() {
+    var drawerButton;
     if (this.screenSizeMediaQuery_.matches) {
       this.element_.classList.add(this.CssClasses_.IS_SMALL_SCREEN);
+      // Hide drawer button and display drawer when moving to a small screen.
```

Comment:
I followed the existing behaviour of MDL lite. When a drawer is used it toggles visibility when the screen sizes are adjusted.  [MDL lite nav layout](https://getmdl.io/components/index.html#layout-section/layout)